ResearchKit/Common/ORKImageCaptureCameraPreviewView.m

```diff
@@ -0,0 +1,187 @@
+    
+        _templateImageView = [[ORKTintedImageView alloc] init];
+        _templateImageView.contentMode = UIViewContentModeScaleAspectFit;
+        _templateImageView.shouldApplyTint = YES;
+        _templateImageView.alpha = 0;
+        [self addSubview:_templateImageView];
+        
+        _capturedImageView = [[ORKTintedImageView alloc] init];
+        _capturedImageView.contentMode = UIViewContentModeScaleAspectFit;
+        [self addSubview:_capturedImageView];
+        
+        NSDictionary *dictionary = NSDictionaryOfVariableBindings(_templateImageView, _capturedImageView);
+        ORKEnableAutoLayoutForViews([dictionary allValues]);
+        // Make the insets for the template image view changable later
+        _templateImageViewTopInset = [NSLayoutConstraint constraintWithItem:_templateImageView attribute: NSLayoutAttributeTop relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeTop multiplier:1 constant:0];
+        [self addConstraint:_templateImageViewTopInset];
+        _templateImageViewLeftInset = [NSLayoutConstraint constraintWithItem:_templateImageView attribute: NSLayoutAttributeLeft relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeLeft multiplier:1 constant:0];
+        [self addConstraint:_templateImageViewLeftInset];
+        _templateImageViewBottomInset = [NSLayoutConstraint constraintWithItem:_templateImageView attribute: NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeBottom multiplier:1 constant:0];
+        [self addConstraint:_templateImageViewBottomInset];
+        _templateImageViewRightInset = [NSLayoutConstraint constraintWithItem:_templateImageView attribute: NSLayoutAttributeRight relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeRight multiplier:1 constant:0];
+        [self addConstraint:_templateImageViewRightInset];
+        // Make the captured image view use the available space (taking into account the layout margins)
+        [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|-[_capturedImageView]-|" options:NSLayoutFormatDirectionLeadingToTrailing metrics:nil views:dictionary]];
+        [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|-[_capturedImageView]-|" options:NSLayoutFormatDirectionLeadingToTrailing metrics:nil views:dictionary]];
+    }
+    return self;
+}
+
+- (AVCaptureSession *)session {
+    return _previewLayer.session;
+}
+
+- (void)setSession:(AVCaptureSession * __nullable)session {
+    _previewLayer.session = session;
+}
+
+- (UIImage *)templateImage {
+    return _templateImageView.image;
+}
+
+- (void)setTemplateImage:(UIImage * __nullable)templateImage {
+    _templateImageView.image = templateImage;
+}
+
+- (void)setHideTemplateImage:(BOOL)hideTemplateImage {
+    _templateImageView.alpha = hideTemplateImage ? 0 : 1;
+}
+
+- (BOOL)hideTemplateImage {
+    return _templateImageView.alpha==0;
+}
```

Comment:
For the above pair of methods:

Can the setter be renamed to setTemplateImageHidden:(BOOL)hidden
The getter would be better if renamed to isTemplateImageHidden